protocol.go

```diff
@@ -35,8 +42,9 @@ const (
 // The remote sends a confirmation after every SCP command, because a failure can occur after every
 // command, the response should be read and checked after sending them.
 type Response struct {
-	Type    ResponseType
-	Message string
+	Type         ResponseType
+	Message      string
+	ProtocolType rune
```

Comment:
What is the reason of using a `rune` type here? A `rune` is 32 bits in size and mostly used for encoding unicode code points. However further down we use it to store a single unsigned byte, so it seems that we are wasting some memory space here.

Moreover, you seem to store the same data in `Type` and `ProtocolType`, is this because you want to differentiate between error messages and protocol messages? Maybe we should simply not store the error status and return it as an actual error from `ParseResponse` (see above).